sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ManifestFileCommitProtocol.scala

```diff
@@ -70,7 +80,24 @@ class ManifestFileCommitProtocol(jobId: String, path: String)
 
   override def abortJob(jobContext: JobContext): Unit = {
     require(fileLog != null, "setupManifestOptions must be called before this function")
-    // Do nothing
+    // best effort cleanup of complete files from failed job
+    // since the file has UUID in its filename, we are safe to try deleting them
+    // not deleting them doesn't help reducing cost to saving files when retrying
+    if (pendingCommitFiles.nonEmpty) {
+      pendingCommitFiles.foreach { file =>
+        val path = new Path(file)
+        val fs = path.getFileSystem(jobContext.getConfiguration)
+        // this is to make sure the file can be seen from driver as well
+        if (fs.exists(path)) {
+          fs.delete(path, false)
```

Comment:
How about adding `try` here to catch `IOException`?
I hope we can iterate the whole `pendingCommitFiles` when we fail to delete some of them.

Reply:
Nice suggestion. I guess we might also not want to make abortJob() failing due to IOException since we are just doing "best-effort". Will update.